plugins/joblib_launcher/hydra_plugins/joblib_launcher/joblib_launcher.py

```diff
@@ -0,0 +1,141 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+import logging
+from pathlib import Path
+from typing import Any, Dict, Optional, Sequence
+
+from joblib import Parallel, delayed  # type: ignore
+from omegaconf import DictConfig, open_dict
+
+from hydra.core.config_loader import ConfigLoader
+from hydra.core.config_search_path import ConfigSearchPath
+from hydra.core.hydra_config import HydraConfig
+from hydra.core.utils import (
+    JobReturn,
+    JobRuntime,
+    configure_log,
+    filter_overrides,
+    run_job,
+    setup_globals,
+)
+from hydra.plugins.launcher import Launcher
+from hydra.plugins.search_path_plugin import SearchPathPlugin
+from hydra.types import TaskFunction
+
+log = logging.getLogger(__name__)
+
+
+class JoblibLauncherSearchPathPlugin(SearchPathPlugin):
+    """
+    This plugin is allowing configuration files provided by the JoblibLauncher plugin to be discovered
+    and used once the JoblibLauncher plugin is installed
+    """
+
+    def manipulate_search_path(self, search_path: ConfigSearchPath) -> None:
+        # Appends the search path for this plugin to the end of the search path
+        search_path.append(
+            "hydra-joblib-launcher", "pkg://hydra_plugins.joblib_launcher_plugin.conf",
+        )
+
+
+class JoblibLauncher(Launcher):
+    def __init__(self, **kwargs_joblib_parallel: Any) -> None:
+        """Parallel Launcher
+
+        Keyword arguments that are not None are passed to joblib.Parallel
+
+        For details, refer to to the Parallel documentation:
+        https://joblib.readthedocs.io/en/latest/generated/joblib.Parallel.html
+        """
+        self.config: Optional[DictConfig] = None
+        self.config_loader: Optional[ConfigLoader] = None
+        self.task_function: Optional[TaskFunction] = None
+
+        self.kwargs_joblib_parallel = {}
```

Comment:
You should be able to do use the config directly, 
```python
Parallel(**kwargs_joblib_parallel)
```

Also, rename **kwargs_joblib_parallel to **jobid_arguments.
I am a bit torn about passing it like this and having another nesting in the config which would allow you to pass in additional things that would be consumed by the plugin itself.
something like:
```yaml
launcher:
    class: hydra_plugins.joblib_launcher.JoblibLauncher
    params:
      thing1: foo # something that goes to the plugin itself.
      jobid: # everything that goes to jobid.
        n_jobs: -1
```